mesos-rxjava-core/src/main/java/org/apache/mesos/rx/java/SinkSubscriber.java

```diff
@@ -0,0 +1,94 @@
+package org.apache.mesos.rx.java;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.netty.util.CharsetUtil;
+import io.reactivex.netty.protocol.http.client.HttpClient;
+import io.reactivex.netty.protocol.http.client.HttpClientRequest;
+import io.reactivex.netty.protocol.http.client.HttpResponseHeaders;
+import org.apache.mesos.v1.scheduler.Protos;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import rx.Observable;
+import rx.Subscriber;
+import rx.exceptions.Exceptions;
+import rx.functions.Func1;
+
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.mesos.rx.java.ProtoUtils.protoToString;
+
+final class SinkSubscriber extends Subscriber<SinkOperation> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SinkSubscriber.class);
+
+    @NotNull
+    private final HttpClient<ByteBuf, ByteBuf> httpClient;
+    @NotNull
+    private final Func1<Protos.Call, Observable<HttpClientRequest<ByteBuf>>> createPost;
```

Comment:
Originally I had that, but then I moved it to be it's own class so that it's not closing over any outer state.  I feel like this is a better approach given that we're trying to promote not closing over state in the functions processing the event stream.

I'll create a follow up PR to more correctly formalize not closing over any external state.